audb/core/load.py

```diff
@@ -535,15 +535,26 @@ def _media(
 ) -> typing.Sequence[str]:
 
     if media is None:
-        media = db.files
-    elif isinstance(media, str):
+        return db.files
+    elif len(media) == 0:
+        return []
+
+    if isinstance(media, str):
         pattern = re.compile(media)
-        media = []
+        requested_media = []
         for m in db.files:
             if pattern.search(m):
-                media.append(m)
+                requested_media.append(m)
+    else:
+        requested_media = media
+
+    if (
+            len(requested_media) == 0
+            or any([file not in db.files for file in requested_media])
```

Comment:
Maybe we should handle this case separately and list only the missing media files?

Reply:
The problem is that you can provide a search string instead of a list of tables.

But we could extend it for the case where the user provides a list.

Reply:
Yes, that's what I meant.

Reply:
Cause if you provide a search string you will never end up in the "or" case I guess.

Reply:
Actually with a search string there is also no need to check again if the files exists. So maybe move the exception handling into the `if - else` condition.

Reply:
I updated it and have no two different error messages depending if a user provided a string or a list.